internal/framework/base/base.go

```diff
@@ -1,55 +1,75 @@
-	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
 	"github.com/hashicorp/terraform-provider-vault/internal/consts"
-	"github.com/hashicorp/terraform-provider-vault/internal/framework/validators"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
 )
 
-// BaseModel describes common fields for all of the Terraform resource data models
-//
-// Ideally this struct would be imbedded into all Resources and DataSources.
-// However, the Terraform Plugin Framework doesn't support unmarshalling nested
-// structs. See https://github.com/hashicorp/terraform-plugin-framework/issues/242
-// So for now, we must duplicate all fields.
-type BaseModel struct {
-	ID        types.String `tfsdk:"id"`
-	Namespace types.String `tfsdk:"namespace"`
+type withMeta struct {
+	meta *provider.ProviderMeta
+}
+
+func (w *withMeta) Meta() *provider.ProviderMeta {
+	return w.meta
+}
+
+// ResourceWithConfigure is a structure to be embedded within a Resource that
+// implements the ResourceWithConfigure interface.
+type ResourceWithConfigure struct {
+	withMeta
+}
+
+// Configure enables provider-level data or clients to be set in the
+// provider-defined Resource type.
+func (r *ResourceWithConfigure) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	if v, ok := request.ProviderData.(*provider.ProviderMeta); ok {
+		r.meta = v
+	}
 }
 
-func baseSchema() map[string]schema.Attribute {
-	return map[string]schema.Attribute{
-		// Required for acceptance testing
-		// https://developer.hashicorp.com/terraform/plugin/framework/acctests#no-id-found-in-attributes
-		"id": schema.StringAttribute{
-			Computed:            true,
-			MarkdownDescription: "ID required by the testing framework",
-		},
-		consts.FieldNamespace: schema.StringAttribute{
-			Optional: true,
-			PlanModifiers: []planmodifier.String{
-				stringplanmodifier.RequiresReplace(),
-			},
-			MarkdownDescription: "Target namespace. (requires Enterprise)",
-			Validators: []validator.String{
-				validators.PathValidator(),
-			},
-		},
+// WithImportByID is intended to be embedded in resources which import state
+// via the "id" attribute.
+//
+// https://developer.hashicorp.com/terraform/plugin/framework/resources/import.
+//
+// This will ensure the Vault namespace is written to state if it is set in the
+// environment.
+// https://registry.terraform.io/providers/hashicorp/vault/latest/docs#namespace-support
+type WithImportByID struct{}
+
+func (w *WithImportByID) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root(consts.FieldID), request, response)
+
+	ns := os.Getenv(consts.EnvVarVaultNamespaceImport)
+	if ns == "" {
```

Comment:
looking at this again, I don't think this is correct. We shouldn't error if the namespace is unset. The resource might not be under a namespace